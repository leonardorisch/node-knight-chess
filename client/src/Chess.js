import React, { Component } from 'react';
import './App.css';

class Chess extends Component {
  constructor(props) {
    super(props);
    this.darkClass = 'dark';
    this.lightClass = 'light';
  }

  render() {
    return (
      <div className="chessTable">
        {this.squareElements()}
      </div>
    );
  }

  squareElements() {
    const squares = []
    for(let x = 1; x < 9; x++) {
      for(let y = 1; y < 9; y++) {
        squares.push(
          <div className={"chessSquare " + this.squareClass(x, y)}
               key={x.toString() + y.toString()}> </div>
        )
      }
    }
    return squares
  }

  squareClass(x, y) {
    const odd = x % 2;

    if (y % 2) {
      return odd ? this.lightClass : this.darkClass
    }

    return odd ? this.darkClass : this.lightClass
  }
}

export default Chess;
