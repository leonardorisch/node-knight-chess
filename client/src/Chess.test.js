import React from 'react';
import Chess from './Chess';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const component = renderer.create(<Chess />)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
