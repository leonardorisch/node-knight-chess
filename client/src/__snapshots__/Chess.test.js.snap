// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="chessTable"
>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
  <div
    className="chessSquare dark"
  >
     
  </div>
  <div
    className="chessSquare light"
  >
     
  </div>
</div>
`;
